src/libraries/System.Private.CoreLib/src/System/Numerics/Vector_1.cs

```diff
@@ -0,0 +1,1507 @@
+            {
+                return value;
+            }
+            else
+            {
+                Vector<T> result = default;
+
+                for (nint index = 0; index < Count; index++)
+                {
+                    result.SetElement(index, ScalarAbs(value.GetElement(index)));
+                }
+
+                return result;
+            }
+        }
+
+        [Intrinsic]
+        internal static unsafe Vector<T> Min(Vector<T> left, Vector<T> right)
+        {
+            Vector<T> result = default;
+
+            for (nint index = 0; index < Count; index++)
+            {
+                T value = ScalarLessThan(left.GetElement(index), right.GetElement(index)) ? left.GetElement(index) : right.GetElement(index);
+                result.SetElement(index, value);
+            }
+
+            return result;
+        }
+
+        [Intrinsic]
+        internal static unsafe Vector<T> Max(Vector<T> left, Vector<T> right)
+        {
+            Vector<T> result = default;
+
+            for (nint index = 0; index < Count; index++)
+            {
+                T value = ScalarGreaterThan(left.GetElement(index), right.GetElement(index)) ? left.GetElement(index) : right.GetElement(index);
+                result.SetElement(index, value);
+            }
+
+            return result;
+        }
+
+        [Intrinsic]
+        internal static T Dot(Vector<T> left, Vector<T> right)
+        {
+            T product = default;
+
+            for (nint index = 0; index < Count; index++)
+            {
+                product = ScalarAdd(product, ScalarMultiply(left.GetElement(index), right.GetElement(index)));
+            }
+
+            return product;
+        }
+
+        [Intrinsic]
+        internal static unsafe Vector<T> SquareRoot(Vector<T> value)
+        {
+            Vector<T> result = default;
+
+            for (nint index = 0; index < Count; index++)
+            {
+                result.SetElement(index, ScalarSqrt(value.GetElement(index)));
+            }
+
+            return result;
+        }
+
+        [Intrinsic]
+        internal static unsafe Vector<T> Ceiling(Vector<T> value)
+        {
+            Vector<T> result = default;
+
+            for (nint index = 0; index < Count; index++)
+            {
+                result.SetElement(index, ScalarCeiling(value.GetElement(index)));
+            }
+
+            return result;
+        }
+
+        [Intrinsic]
+        internal static unsafe Vector<T> Floor(Vector<T> value)
+        {
+            Vector<T> result = default;
+
+            for (nint index = 0; index < Count; index++)
+            {
+                result.SetElement(index, ScalarFloor(value.GetElement(index)));
+            }
+
+            return result;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool ScalarEquals(T left, T right)
+        {
+            if (typeof(T) == typeof(byte))
```

Comment:
This is where the actual per-type logic lives, same as previously for most of the methods.